modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/resource/v1_0/KeywordResourceImpl.java

```diff
@@ -417,7 +420,25 @@ private Page<Keyword> _getKeywordsPage(
 				searchContext.addVulcanAggregation(aggregation);
 				searchContext.setAttribute(Field.NAME, search);
 				searchContext.setCompanyId(contextCompany.getCompanyId());
-				searchContext.setGroupIds(new long[] {groupId});
+
+				DepotEntry depotEntry = _depotEntryService.fetchGroupDepotEntry(
+					groupId);
+
+				if ((depotEntry != null) &&
+					(depotEntry.getType() == DepotConstants.TYPE_SPACE)) {
+
+					searchContext.setAttribute(
+						"assetTagGroupRelsGroupIds",
+						new long[] {
+							groupId, GroupConstants.ANY_PARENT_GROUP_ID
+						});
+				}
+				else {
+					searchContext.setGroupIds(
+						new long[] {
+							groupId, GroupConstants.ANY_PARENT_GROUP_ID
```

Comment:
we don't need to include `GroupConstants.ANY_PARENT_GROUP_ID` here since no assetTag will ever be added with this groupId